File: Python/textbook_track/chapter01/ba1a.py
# Compute the Number of Times a Pattern Appears in a Text
import sys


def count_pattern(text, pattern):
    pattern_length = len(pattern)
    count = 0
    for ix in range(len(text)-pattern_length+1):
        substring = text[ix:ix+pattern_length]
        if substring == pattern:
            count += 1
    return count


def main():
    text, pattern = sys.stdin.read().splitlines()
    result = count_pattern(text, pattern)
    print(result)


if __name__ == '__main__':
    main()

File: Julia/textbook_track/chapter01/ba1a.jl
# Compute the Number of Times a Pattern Appears in a Text

function count_pattern(text, pattern)
    k = length(pattern)
    count = 0
    for ix in 1:(length(text)-k+1)
        substring = text[ix:ix+k-1]
        if substring == pattern
            count += 1
        end
    end
    count
end


function main()
    text = readline()
    pattern = readline()
    result = count_pattern(text, pattern)
    println(result)
end


main()
